codegens/curl/lib/util.js

```diff
@@ -199,5 +201,77 @@ var self = module.exports = {
         contentType: contentType
       });
     }
+  },
+
+  /**
+   * @param {Object} body
+   * @returns {boolean}
+   *
+   * Determines if a request body is actually empty.
+   * This is needed because body.isEmpty() returns false for formdata
+   * and urlencoded when they contain only disabled params which will not
+   * be a part of the curl request.
+   */
+  isBodyEmpty (body) {
+    if (!body) {
+      return true;
+    }
+
+    if (body.isEmpty()) {
+      return true;
+    }
+
+    if (body.mode === 'formdata' || body.mode === 'urlencoded') {
+      let memberCount = 0;
+      body[body.mode].members && body[body.mode].members.forEach((param) => {
+        if (!param.disabled) {
+          memberCount += 1;
+        }
+      });
+
+      return memberCount === 0;
+    }
+
+    return false;
+  },
+
+  /**
+   * Decide whether we should add the HTTP method explicitly to the cURL command.
+   *
+   * @param {Object} request
+   * @param {Object} options
+   *
+   * @returns {Boolean}
+   */
+  shouldAddHttpMethod: function (request, options) {
+    const followOriginalHttpMethod =
+      _.get(request, 'protocolProfileBehavior.followOriginalHttpMethod', options.followOriginalHttpMethod),
```

Comment:
@dhwaneetbhatt Not sure here, Shouldn't option override the behaviour of what's present in request if present?

Reply:
@VShingala The default value of the option is always `false`. There is no undefined state, so we cannot determine if the user or the default value sets the option's current value.

I can change this to `_.get(request, 'protocolProfileBehavior.followOriginalHttpMethod', false) || options.followOriginalHttpMethod` to honour the option too. But it won't be an "override". Any better ideas?

Reply:
So my main intention was to clarify whether `protocolProfileBehavior.followOriginalHttpMethod` should take precedence or `options.followOriginalHttpMethod`. Current _.get statement is fine, just what should be first followed is something we can discuss.

Reply:
As per the discussion, the snippet-level option overrides the request-level option. Let me re-think the condition here. Thanks for initiating this discussion.